### PR TITLE
uncomment farm code

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

File: locked-asset/proxy_dex/src/proxy_pair.rs [hunk 1]
```diff
@@ -64,25 +61,26 @@ pub trait ProxyPairModule:
...
                 WrappedLpToken::new_from_payments(&payments, &wrapped_lp_mapper);
 
+            self.burn_multi_esdt(&payments);
+
             self.merge_wrapped_lp_tokens_with_virtual_pos(wrapped_lp_tokens, new_token_attributes)
-                .attributes
         } else {
-            new_token_attributes
-        };
+            let new_token_amount = new_token_attributes.get_total_supply().clone();
```
Comment (sasurobert) [inline locked-asset/proxy_dex/src/proxy_pair.rs:76]: don't we need to burn in this case as well ?

**Review by sasurobert — APPROVED** [review]

File: locked-asset/proxy_dex/src/proxy_pair.rs (hunk 1 above)
Comment (dorin-iancu) [reply locked-asset/proxy_dex/src/proxy_pair.rs:76]: Nope, the initial payments are Locked tokens, and another token. We need to keep the locked tokens to send them back on Remove Liquidity, and the other payment, is used for Add Liquidity.

**Review by sasurobert — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]

File: dex/farm/src/lib.rs [hunk 2]
```diff
@@ -151,19 +149,7 @@ pub trait Farm:
...
-        let payments = self.call_value().all_esdt_transfers();
-
-        let attrs = self.get_default_merged_farm_token_attributes(&payments, Option::None);
-        let farm_token_id = self.farm_token().get_token_id();
-        self.burn_farm_tokens_from_payments(&payments);
-
-        let new_tokens =
-            self.mint_farm_tokens(farm_token_id, attrs.current_farm_amount.clone(), &attrs);
-
+    fn merge_farm_tokens_endpoint(&self) -> EsdtTokenPayment<Self::Api> {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply dex/farm/src/lib.rs:152]: it was not used ?

File: locked-asset/proxy_dex/src/wrapped_farm_attributes.rs [hunk 3]
```diff
@@ -139,8 +139,7 @@ pub fn merge_wrapped_farm_tokens<M: CallTypeApi + StorageMapperApi>(
         )
     };
 
-    let merged_farm_tokens =
-        merge_farm_tokens_through_farm(original_caller, farm_address, farm_tokens_to_merge);
+    let merged_farm_tokens = merge_farm_tokens_through_farm(farm_address, farm_tokens_to_merge);
```
Comment (sasurobert) [inline locked-asset/proxy_dex/src/wrapped_farm_attributes.rs:142]: In case of merge do we recompute the energy ? If not, we might have some issue.

Comment (dorin-iancu) [reply locked-asset/proxy_dex/src/wrapped_farm_attributes.rs:142]: Nope, we do not recompute in case of plain merging.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**